cloverage/src/cloverage/rewrite.clj

```diff
@@ -0,0 +1,54 @@
+(ns cloverage.rewrite)
+
+(defn better-empty [x]
+  (with-meta
+    (if (instance? clojure.lang.IMapEntry x)
+      []
+      (empty x))
+    (meta x)))
+
+(defn better-into [e s]
+  (if (list? e)
+    (with-meta
+      (seq s)
+      (meta e))
+    (into e s)))
+
+(defn walk
+  [inner outer form]
+  (if (coll? form)
+    (let [e (better-empty form)
+          contents (map inner form)]
+      (outer (better-into e contents)))
+    (outer form)))
```

Comment:
I had to write my own walk code because the walker in clojure doesn't preserve meta. If there's a less silly way to do this, let me know.